test/test_expanded_weights.py

```diff
@@ -193,6 +193,16 @@ def test_expanded_weight_per_sample_grad_mean(self, device, dtype, op):
 
             self._compare_ew_and_for_loop_per_sample_grads(op, sample_input, torch.mean)
 
+    @ops(filter(lambda op: op.supports_expanded_weight, op_db), dtypes=OpDTypes.supported, allowed_dtypes=(torch.double,))
+    def test_expanded_weights_per_sample_grad_input_no_grad(self, device, dtype, op):
+        sample_inputs = op.sample_inputs(device, dtype, requires_grad=True)
+        for sample_input in supported_inputs(op, sample_inputs):
+            if op.name == "nn.functional.embedding":  # embedding flips its argument order for autograd tests
+                sample_input = SampleInput(sample_input.args[0], args=(sample_input.input,), kwargs=sample_input.kwargs)
+            sample_input.input.requires_grad_(False)
```

Comment:
For this, no because that will make all the arguments, including weights, have requires_grad=False. We still want the weights to require gradients so that we test Expanded Weights backwards but we just don't want the input to have a gradient to test the non-short circuited pathway (it checks if input requires gradient since we might be able to avoid saving state if the input doesn't require gradients)

Reply:
Gotcha, thanks for the explanation